### PR TITLE
daemon: add option to skip CRD creation

### LINKED ISSUE #7434: Add option to skip auto-creating CRDs
Some users have asked for the ability to disable the auto-creation of CRDs and to defer such creation to the user. This in turn would allow to strip several privileges from the RBAC requirements.

### PR DESCRIPTION
**Pull request by charlyx**
We've added a new option called `--skip-crd-creation` that skips automatic CRD creation and let users create it themselves.
We've also added an example of such a creation to help them make it.

Fixes: #7434

Signed-off-by: Pierre-Yves AILLET <pyaillet@gmail.com>
Signed-off-by: Charles-Henri GUERIN <charleshenri.guerin@icloud.com>

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/7989)

### CODE REVIEW

File: daemon/daemon_main.go [hunk 1]
```diff
@@ -737,6 +737,9 @@ func init() {
 	flags.MarkHidden(option.SelectiveRegeneration)
 	option.BindEnv(option.SelectiveRegeneration)
 
+	flags.Bool(option.SkipCRDCreation, false, "skip CRD creation")
```
Comment (aanm) [inline daemon/daemon_main.go]: `s/skip CRD creation/Skip Kubernetes Custom Resource Definitions creations`

File: examples/crds/ciliumnetworkpolicies.yaml [hunk 2]
```diff
@@ -0,0 +1,4020 @@
...
+        LabelSelectorRequirement:
+          description: A label selector requirement is a selector that contains values,
+            a key, and an operator that relates the key and values.
+          properties:
+            key:
+              description: key is the label key that the selector applies to.
+              type: string
+            operator:
+              description: operator represents a key's relationship to a set of values.
+                Valid operators are In, NotIn, Exists and DoesNotE
```
Comment (aanm) [inline examples/crds/ciliumnetworkpolicies.yaml]: It's missing a new line

File: examples/crds/ciliumendpoints.yaml [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+  scope: Namespaced
+  subresources:
+    status: {}
+  validation:
+    openAPIV3Schema: {}
+  version: v2
+  versions:
+  - name: v2
+    served: true
+    storage: true
```
Comment (aanm) [inline examples/crds/ciliumendpoints.yaml]: It's missing a new line

**Review by aanm — CHANGES_REQUESTED** [review]
Thank you for the changes, they look great

I would put the `examples/crds` files in the `examples/kubernetes/addons/crds` directory.

File: Documentation/cmdref/cilium-agent.md [hunk 4]
```diff
@@ -111,6 +111,7 @@ cilium-agent [flags]
       --restore                                    Restores state, if possible, from previous daemon (default true)
       --sidecar-istio-proxy-image string           Regular expression matching compatible Istio sidecar istio-proxy container image names (default "cilium/istio_proxy")
       --single-cluster-route                       Use a single cluster route instead of per node routes
+      --skip-crd-creation                          skip CRD creation
```
Comment (aanm) [inline Documentation/cmdref/cilium-agent.md]: you will need to generate this doc again `make -C Documentation cmdref`

**Review by aanm — CHANGES_REQUESTED** [review]

**Comment by aanm** [comment]
test-me-please

**Review by tgraf — APPROVED** [review]
I'm fine with the location for now. Maybe we should consider moving all installation yamls to `install/` to make it clearer what are examples and what are installation artifacts.

**Comment by tgraf** [comment]
@charlyx Please rebase against master. @aanm will do a final review.

**Comment by charlyx** [comment]
@tgraf @aanm the branch has been rebased against master 👌

**Comment by tgraf** [comment]
test-me-please

**Review by aanm — APPROVED** [review]

**Comment by tgraf** [comment]
Thanks @charlyx!

**Pull request merged**